src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -473,7 +474,156 @@ pass::EliminateSqueeze::EliminateSqueeze() {
         return false;
     };
 
-    auto m = std::make_shared<pattern::Matcher>(squeeze_pattern, matcher_name);
+    auto m = make_shared<pattern::Matcher>(squeeze_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+namespace {
+bool check_squeeze(const shared_ptr<Node>& node) {
+    auto squeeze = dynamic_pointer_cast<ov::opset9::Squeeze>(node);
+    if (squeeze) {
+        auto axis = dynamic_pointer_cast<ov::opset9::Constant>(squeeze->input_value(1).get_node_shared_ptr());
+        if (axis) {
+            auto axis_val = axis->cast_vector<int64_t>();
+            if (axis_val.size() == 1 && axis_val[0] == 1) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool check_reshape(const shared_ptr<Node>& node) {
+    auto reshape = dynamic_pointer_cast<ov::opset9::Reshape>(node);
+    if (reshape) {
+        auto shape_pattern = dynamic_pointer_cast<ov::opset9::Constant>(reshape->input_value(1).get_node_shared_ptr());
+        if (shape_pattern) {
+            auto pattern_val = shape_pattern->cast_vector<int64_t>();
+            pattern_val.insert(pattern_val.begin() + 1, 1);
+            auto in_shape = reshape->input_value(0).get_partial_shape();
+            if (in_shape.compatible(pattern_val)) {
```

Comment:
Not sure what we try to achieve here. As I understand -- you try to check if this reshape is a simple Unsqueeze of axis 1, right? Since Reshape has possibility to have 0 and -1 pattern values -- I think that casting it to PartialShape could result in unpredictable results as -1 means dynamic dimension and dynamic dimension is compatible with everything. Let's think how we can check it the other way.

Reply:
Added additional check for 0 and -1